css/src/components/layout.scss

```diff
@@ -0,0 +1,110 @@
+$side-bar-width-wide: 4fr !default;
+
+.layout {
+	display: flex;
+	flex-direction: column;
+	min-height: 100vh;
+	overflow-x: hidden;
+
+	// --layout-gutter by default, see tokens/layout.scss
+	#{$layout-gap-custom-property-name}: $layout-gap;
+	#{$layout-gap-scalable-custom-property-name}: $layout-gap;
+
+	@include widescreen {
+		#{$layout-gap-scalable-custom-property-name}: $layout-widescreen-gap;
+	}
+}
+
+.layout-body {
+	max-width: 100%;
+	overflow: hidden;
```

Comment:
I'm a little nervous about having `overflow` stuff applied (here and the `.layout`) - `overflow` parents are one of the things that can prevent `position: sticky` from working as expected.

Reply:
Unfortunately at some point up the DOM chain overflow is absolutely necessary on sites with large chunks of unbroken text (i.e. code blocks in our case(s)). I think this is necessary here.

Reply:
I'm not sure I'm following - surely in those cases, the `overflow` would get applied to the child that's causing overflow issues.

As it stands, I'm pretty certain applying this here would, for instance, break our Conceptual right rail or our TOC functionality. Not sure we can proceed with that.

Reply:
Let's test it out! I have a POC, pinging ya!

Reply:
Just verified with the PR in docs-ui: this layout code prevents the TOC and right rail from being sticky/following you down the page:

![Test page, scrolled down a little. The tops of the table of contents and the right rail have scrolled off the top of the page.](https://github.com/microsoft/atlas-design/assets/18060369/02919c6b-8412-44c6-b0ce-ec03156647cf)

Reply:
Should have just updated with a fix that should do the trick. Please have a peek again!

Reply:
Retested, and this all looks great to me now!